lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -4,68 +4,54 @@ class Deployment < KubernetesResource
     TIMEOUT = 5.minutes
 
     def sync
-      json_data, _err, st = kubectl.run("get", type, @name, "--output=json")
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
       @found = st.success?
-      @rollout_data = {}
+      @rollout_data = { "replicas" => 0 }
       @status = nil
-      @representative_pod = nil
-      @pods = []
+      @latest_rs = nil
 
       if @found
-        @rollout_data = JSON.parse(json_data)["status"]
-          .slice("updatedReplicas", "replicas", "availableReplicas", "unavailableReplicas")
-        @status = @rollout_data.map { |replica_state, num| "#{num} #{replica_state}" }.join(", ")
+        deployment_data = JSON.parse(raw_json)
+        latest_rs_data = get_latest_rs(deployment_data)
+        @latest_rs = ReplicaSet.new(
+          name: latest_rs_data["metadata"]["name"],
+          namespace: namespace,
+          context: context,
+          file: nil,
+          parent: "#{@name.capitalize} deployment",
+          logger: @logger
+        )
+        @latest_rs.interpret_json_data(latest_rs_data)
+        @latest_rs.deploy_started = @deploy_started
 
-        pod_list, _err, st = kubectl.run("get", "pods", "-a", "-l", "name=#{name}", "--output=json")
-        if st.success?
-          pods_json = JSON.parse(pod_list)["items"]
-          pods_json.each do |pod_json|
-            pod_name = pod_json["metadata"]["name"]
-            pod = Pod.new(
-              name: pod_name,
-              namespace: namespace,
-              context: context,
-              file: nil,
-              parent: "#{@name.capitalize} deployment",
-              logger: @logger
-            )
-            pod.deploy_started = @deploy_started
-            pod.interpret_json_data(pod_json)
-
-            if !@representative_pod && pod_probably_new?(pod_json)
-              @representative_pod = pod
-            end
-            @pods << pod
-          end
-        end
+        @rollout_data.merge!(deployment_data["status"]
+          .slice("replicas", "updatedReplicas", "availableReplicas", "unavailableReplicas"))
+        @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
       end
     end
 
-    def fetch_logs
-      @representative_pod ? @representative_pod.fetch_logs : {}
+    def fetch_events
+      return {} unless @latest_rs.present?
+      @latest_rs.fetch_events
     end
 
-    def fetch_events
-      own_events = super
-      return own_events unless @representative_pod
-      own_events.merge(@representative_pod.fetch_events)
+    def fetch_logs
+      return {} unless @latest_rs.present?
+      @latest_rs.fetch_logs
```

Comment:
`kubectl get logs deployment/web -c nginx` (or whatever) is a valid command. However, looking at the source, I  think it might favour pods from the old RS under some circumstances, and certainly won't exclude them. https://github.com/kubernetes/kubernetes/blob/dbd1503b6546b0b6048787bad0e63515f3281506/pkg/controller/controller_utils.go#L669-L698